server/src/service-layer/response-models/UserResponse.ts

```diff
@@ -14,11 +14,11 @@ type CombinedUserData = UserAdditionalInfo & {
   /**
    * The email the user uses to log in
    */
-  email: string
+  email?: string
```

Comment:
id recommend not making this change in here as this this might be enforced for a reason

Reply:
they are undefined (and giving errors) because the email and membership are handled separately than in the `UserDataService`, what you would need to do is use the function `bulkRetrieveUsersByUid` in `AuthService` to get that data,

see the `getAllUsers` method in `AdminController` for how the email is retrieved.

Reply:
also small tip, you can do `Partial<{key1:string}>` if you want to make **ALL** fields of a type optional